README.md

```diff
@@ -1,3 +1,190 @@
+- [License](#license)
+
+
+## Security
+
+It should be noted that this project requires the use of some "secret" variables that should be tailored to your development environment and never committed to the repository. These variables are read by [dotenv](https://www.npmjs.com/package/dotenv) and should be defined in a file called `.env` in the project root.
+
+After cloning the repository, you should duplicate and rename `.env.example` to `.env`, then tailor the variables to your local environment accordingly:
+
+```bash
+$ cp .env.example .env
+$ vi .env
+```
+
+`.env.example` has suitable local development defaults for most variables, so you should only have to set any missing values (i.e. `JWT_SECRET` & `SIGNER_PRIVATE_KEY`.) Descriptions for each variable are listed in [.env.example](.env.example).
+
+## Install
+
+There's a lot involved in setting up this project, so bear with me.
+
+### Clone & Set Up Required Repositories
+
+1. First, clone this repository and all dependency repositories (see below for more details):
+
+```bash
+$ git clone https://github.com/codex-protocol/service.codex-registry-api
+$ git clone https://github.com/codex-protocol/contract.codex-registry
+$ git clone https://github.com/codex-protocol/npm.ethereum-service
+$ git clone https://github.com/codex-protocol/service.eel
+```
+
+**IMPORTANT NOTE:** It's necessary to have all of these repositories in the same directory, since our npm scripts assume this is the case.
+
+1. Then install all npm packages in each repository:
+
+```bash
+$ cd service.codex-registry-api
+$ npm install
+
+$ cd ../contract.codex-registry
+$ npm install
+
+$ cd ../npm.ethereum-service
+$ npm install
+
+$ cd ../service.eel
+$ npm install
+```
+
+2. After you've installed all npm packages, you will also need to [npm link](https://docs.npmjs.com/cli/link) the ethereum-service repository so that the API can use your locally-deployed smart contracts:
+
+```bash
+$ cd npm.ethereum-service
+$ npm link
+
+$ cd ../service.codex-registry-api
+$ npm link @codex-protocol/ethereum-service
+```
+
+Now when you deploy the smart contracts locally, the API will be able to pull the ABIs from the linked ethereum-service repository.
+
+**IMPORTANT NOTE:** every time you run `npm install` in the API repository, you will need to re-link the ethereum-service repository. For convenience, you can simply run the npm script `npm run link-all` to link, or `npm run install-and-link` to install and link in one step.
+
+### Dependencies
+
+Now you will need to install & set up some dependencies.
+
+1. [MongoDB](https://www.mongodb.com/download-center)
+
+MongoDB is where off-chain data (e.g. Codex Record metadata) is stored by the API. It's also used by [Agenda](https://www.npmjs.com/package/agenda) to track [jobs](src/jobs).
+
+MongoDB can be installed via [Homebrew](https://brew.sh/):
+
+```bash
+$ brew install mongodb
+```
+
+or downloaded directly from [the MongoDB website](https://www.mongodb.com/download-center).
+
```

Comment:
A simple link to download mongo might be enough. listing `brew install ...` assumes Mac OS environment.

Reply:
Yeah our whole setup kind of assumes you're on a unix-like environment. That's been on the back of my mind for a while now. If you tried to run any of the npm `copy` scripts, for example, they would fail because we use `cp` and not some platform-independent tool [like this.](https://www.npmjs.com/package/cp).

I want to leave the brew examples, but I will clarify that that's for Mac OS.